### ISSUE TITLE
RowsEvent Header is nil

### ISSUE THREAD

**Issue by uudashr** [go-mysql-org/go-mysql#629]
While using canal
```
c.Run()
```

it will try to get the log from beginning, and turns out the Header is nil.
But for new event, the header is not nil.

**Comment by dveeden** [comment]
Yes the header can be nil. I think your code should test for that.

Feel free to re-open or create a new issue if there is more to this.
